### ISSUE TITLE
Found addresses duplicates

### ISSUE THREAD

**Issue by mgrinzPlayer** [cheat-engine/cheat-engine#1018]
A bug introduced in PR #989 

Test script:
```Lua
local result = AOBScan("01 02 03","*W*X-C")
local duplicatesTest = {}

for i=0,result.Count-1 do
  local addr = result[i]
  duplicatesTest[addr] = (duplicatesTest[addr] or 0) + 1
end

local duplicateFound = false

for k,v in pairs(duplicatesTest) do
  if v>=2 then duplicateFound=true break end
end

if duplicateFound then print("There are address duplicates")
else                   print("No address duplicates. All ok") end
```

**Comment by cheat-engine** [comment]
should be fixed now

**Comment by mgrinzPlayer** [comment]
Yes. But there's a little problem. Now progressbar doesn't move...

**Comment by cheat-engine** [comment]
Try lowering your buffersize in settings. The progressbar only updates after a whole readprocessmemory block has finished scanning

**Comment by mgrinzPlayer** [comment]
Size of scanbuffer is 512KB (default settings)

![obraz](https://user-images.githubusercontent.com/9157726/70912752-ed3f3a00-2014-11ea-81fa-7cf70cde6ddc.png)
![obraz](https://user-images.githubusercontent.com/9157726/70912762-f203ee00-2014-11ea-8de2-86cc42487123.png)

CE7.0 shows progressbar animation, CE7.1 doesn't. Scan takes the same amount of time.


PS: this block merging should increase scan speed right?
But I do not see any improvement in CE7.1. I tested it with BL3 process and 19 aobscanmodules in one AA.
It takes 32 seconds. In CE7.0 it takes the same amount of time.
(note: without Lua extensions, I renamed the autorun folder. Just to be sure.)
